src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,275 @@
+	VertiqSerialInterface *_serial_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
+	EntryWrapper<float, float> _voltage_max_entry;
+	EntryWrapper<uint8_t, int32_t> _control_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _motor_direction_entry;
+	EntryWrapper<uint8_t, int32_t> _fc_direction_entry;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _throttle_cvi_entry;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _pulsing_voltage_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _x_cvi_entry;
+	EntryWrapper<uint8_t, int32_t> _y_cvi_entry;
+	EntryWrapper<float, float> _pulse_zero_angle_entry;
+	EntryWrapper<float, float> _pulse_velo_cutoff_entry;
+	EntryWrapper<float, float> _pulse_torque_offset_entry;
+	EntryWrapper<float, float> _pulse_volt_limit_entry;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq client information
+	//Some constants to help us out
+	static const uint8_t MAXIMUM_CONFIGURATION_CLIENTS = 20; //These are clients whose module ID will change when Target Module ID changes
+	static const uint8_t MAXIMUM_OPERATIONAL_CLIENTS = 20; //These are clients that are used for module control/telemetry. They have a static Module ID
+
+	//Client Arrays
+	ClientAbstract *_configuration_client_array[MAXIMUM_CONFIGURATION_CLIENTS];
+	uint8_t _configuration_clients_in_use = 0;
+
+	ClientAbstract *_operational_client_array[MAXIMUM_OPERATIONAL_CLIENTS];
+	uint8_t _operational_clients_in_use = 0;
+
+	//Clients
+	PropellerMotorControlClient _broadcast_prop_motor_control;
+	ArmingHandlerClient _broadcast_arming_handler;
+	IQUartFlightControllerInterfaceClient *_telem_ifci;
+	EscPropellerInputParserClient *_prop_input_parser_client;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	//Make all of the clients that we need to talk to the IFCI config params
+	IQUartFlightControllerInterfaceClient *_ifci_client;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	VoltageSuperPositionClient *_voltage_superposition_client;
+	PulsingRectangularInputParserClient *_pulsing_rectangular_input_parser_client;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq Client Entry information
+	static const uint8_t MAX_CLIENT_ENTRIES = 40;
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	uint8_t _added_entry_wrappers = 13;
+	AbstractEntryWrapper *_entry_wrappers[MAX_CLIENT_ENTRIES] = {&_velocity_max_entry, &_voltage_max_entry, &_pulse_zero_angle_entry, &_pulse_velo_cutoff_entry, &_pulse_torque_offset_entry, &_pulse_volt_limit_entry, &_control_mode_entry, &_motor_direction_entry, &_fc_direction_entry, &_throttle_cvi_entry, &_pulsing_voltage_mode_entry, &_x_cvi_entry, &_y_cvi_entry};
```

Comment:
You make static lists of the entry wrappers here, though you make them of size MAX_CLIENT_ENTRIES, so they have plenty of extra space to add new client entries. But that means you need to be looking for the configuration defines in multiple places between the header and the source  and you have made a bunch of class level entry variables. And you still need to call ConfigureStruct on each of them individually in the source file

You have defined the AddClientEntry function above for how to just add in any arbitrary entry to this list though, and you make use of it in the extensibility demo to just generically add the entries you want in the source file. Might it be more consistent and cleaner to just make a blank array of entries, and add each of these entries with AddClientEntry in the source file instead of making them into a list here and calling ConfigureStruct on each of them in the source file? That way you would not need the class variables for each entry, you don't need these initializer lists, and you don't need to call ConfigureStruct on each of them, the AddClientEntry does that already. Also makes the way you add entries consistent with the way that you want customers to add new entries, following along with your extensibility demo.